service/src/main/java/bio/terra/catalog/service/dataset/exception/DatasetNotFoundException.java

```diff
@@ -0,0 +1,9 @@
+package bio.terra.catalog.service.dataset.exception;
+
+import bio.terra.common.exception.NotFoundException;
+
+public class DatasetNotFoundException extends NotFoundException {
```

Comment:
The sonarcloud check fails because these exceptions have 6 parent classes! I'm wondering if I should just use the parent class instead @pshapiro4broad?

Reply:
I think this can be fixed by customizing the scan to treat `ErrorReportException` as the root of the inheritance tree. I'll look at doing that now.